sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -776,11 +777,11 @@ private static Coder resolveCoder(Class deserializer) {
               continue;
             }
             if (returnType.equals(byte[].class)) {
-              return ByteArrayCoder.of();
+              return NullableCoder.of(ByteArrayCoder.of());
             } else if (returnType.equals(Integer.class)) {
-              return VarIntCoder.of();
+              return NullableCoder.of(VarIntCoder.of());
             } else if (returnType.equals(Long.class)) {
-              return VarLongCoder.of();
+              return NullableCoder.of(VarLongCoder.of());
```

Comment:
Is this going to break the Go Kafka connector?

CC: @youngoli

Reply:
:( sorry, Go didn't have a kafkaio when BEAM-10529 was filed.

Reply:
Probably we can introduce a new transform that is only exposed through the Python API ?

Reply:
Another option is to update URN_WITH_METADATA transforms to support nulls: https://github.com/apache/beam/blob/90dd44e81b14bd5a35f808c8962cf22cd69497c8/sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java#L803

Seems like Go SDK uses "URN_WITHOUT_METADATA" transform: https://github.com/apache/beam/blob/90dd44e81b14bd5a35f808c8962cf22cd69497c8/sdks/go/pkg/beam/io/xlang/kafkaio/kafka.go#L84

Reply:
I think exposing the Go Nullable coder is a good option. That way we have it in the future, and we don't need to introduce work arounds in our code

Reply:
Agreed that would be the best solution, but to be clear, are you planning on doing that for the next step on this PR?

Reply:
Yep. I'm working on that now. I'm unfamiliar with Go, but making good progress so far

Reply:
Awesome! Thank you for taking this on